sql/archive.sql

```diff
@@ -0,0 +1,48 @@
+SET statement_timeout=0;
+
+-- Move old contracts to archive tables
+DO $$
+DECLARE
+    contract_ids UUID[];
+    link_ids UUID[];
+    primary_link_ids: UUID[];
+    secondary_link_ids: UUID[];
+BEGIN
+    LOOP
+        -- Select contracts we no longer need in the main table
+        SELECT array_agg(id) INTO contract_ids FROM (
+            SELECT id FROM cards WHERE (type='execute@1.0.0' OR (type='session@1.0.0' AND slug!='session-admin-kernel') OR type='external-event@1.0.0' OR active=false)
+                AND created_at < to_char(CURRENT_DATE - INTERVAL '3 months', 'YYYY-MM-DD')::date LIMIT 1000
+        ) as subset;
+        IF (contract_ids IS NULL) THEN
+            RAISE INFO 'No more matching contracts found, quitting...';
+            EXIT;
+        END IF;
+        RAISE INFO 'Archiving % contracts...', array_length(contract_ids, 1);
+
+        -- Select links for matching contracts
+        SELECT array_agg(id) INTO primary_link_ids FROM links2 WHERE fromid = ANY(contract_ids) OR toid = ANY(contract_ids);
+        SELECT array_agg(id) INTO secondary_link_ids FROM links2 WHERE fromid = ANY(primary_link_ids) OR toid = ANY(primary_link_ids);
+        link_ids = array_cat(primary_link_ids, secondary_link_ids);
+        RAISE INFO 'Archiving % links...', array_length(link_ids, 1);
```

Comment:
@ramirogm @Page- Tested this function on a very small subset of session contracts from 2018/2019 and ran into an issue. Found that there are links linked to other links, which we need to account for. From what I can tell, this happens when, for example, `contract-a` is manually linked `contract-b` by a user. This action results in a `create` contract that is linked by `link-a` to `link-b` which in turn is linking `contract-a` to `contract-b`. A bit confusing, but I think my SQL accounts for this. I also don't think there will be a third layer to this, but won't really know until I run the SQL. Looking for a sanity check!